### DISCUSSION TITLE
sml cmdtelegram in variable?

### DISCUSSION THREAD

**Discussion by nikito7** [arendst/Tasmota#11441] [Q&A] [answered]
Is this possible?
Can be done?

```js
>D

raw=r010400010001,r0104006C0007,r010400730007,r0104007B0005,r010400260003,r010400160006

>BS

=>sensor53 r

>S

if something
then
raw=r010400010001
=>sensor53 r
endif

if something2
then
raw=r0104006C0007,r010400730007,r0104007B0005,r010400260003,r010400160006
=>sensor53 r
endif

>M 1

+1,14,m,1,9600,EB1,5,12,%raw%

```

**Comment by gemu2015** [comment, marked as answer]
this might be possible but not as you typed. (never tested though)
give it a try

raw must be a string and strings can only have 48 chars per default.
so you must expand the maximum allowed string size by e.g.
#define SCRIPT_MAXSSIZE 100
#define SML_REPLACE_VARS

then
\>D 100
raw="r010400010001,r0104006C0007,r010400730007,r0104007B0005,r010400260003,r010400160006"

\>S

if something
then
raw="r010400010001"
=>sensor53 r
endif

if something2
then
raw="r0104006C0007,r010400730007,r0104007B0005,r010400260003,r010400160006"
=>sensor53 r
endif

\>M 1
+1,14,m,1,9600,EB1,5,12,%raw%

**Reply by nikito7** [reply]
Don't work.
SML_REPLACE_VARS only work in decoders

I tryed sometime ago change "loop interval", it's just a number and didn't worked...

**Reply by gemu2015** [reply]
you are right, works only in decode lines.
What do you want to accomplish with  this switching ?

**Reply by nikito7** [reply]
Get total energy less times

**Reply by nikito7** [reply]
Or even do: one time resquest cmd

**Reply by gemu2015** [reply]
Why ???

**Reply by nikito7** [reply]
It's not for me. I'm ok spamming modbus 😁

Someone wants:

Voltage every 5seconds
Total kwh every 1hour

🤣

**Reply by gemu2015** [reply]
you mean over MQTT ?
you may switch off MQTT  by choosing * for MQTT designator
+1,14,m,1,9600,*

and transmit the MQTT JSON yourself e.g. every hour

**Reply by nikito7** [reply]
No. Modbus requests.

If you have 10 single registers. 5s delay.

Loop will have 50 seconds

**Comment by nikito7** [comment]
```js
>D 100

raw="r010400010001"

>B

smlj=0
tper=30

=>sensor53 r

>S

if upsecs==10
then
raw="r010400010001,r0104006C0007,r010400730007,r0104007B0005,r010400260003,r010400160006"
=>sensor53 r
endif

if upsecs==25
then
smlj=1
endif

if upsecs==30
then
raw="r010400010001,r0104006C0007,r010400730007,r0104007B0005"
=>sensor53 r
endif

print test4: raw %raw%

>M 1

```



Result:

```
18:33:31.378 test4: raw ???
18:33:32.380 test4: raw ???
```
